lib/publiccloud/ecr.pm

```diff
@@ -0,0 +1,44 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper class for Amazon Elastic Container Registry (ECR)
+#
+# Maintainer: Ivan Lausuch <ilausuch@suse.de>, qa-c team <qa-c@suse.de>
+# Documentation: https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html
+
+package publiccloud::ecr;
+use Mojo::Base 'publiccloud::aws';
+use testapi;
+
+sub push_container_img {
+    my ($self, $name) = @_;
+
+    my $aws_account_id = $self->get_user_id();
+    my $region         = get_var('PUBLIC_CLOUD_REGION',                'eu-central-1');
+    my $repository     = get_var("PUBLIC_CLOUD_CONTAINER_IMAGES_REPO", 'suse-qec-testing');
+    my $tag            = (split /-/, get_required_var("NAME"))[0];
```

Comment:
Using `get_required_var('NAME')` here is not a good Idea, I would like to avoid this!

Reply:
Why not is a good idea? I am interested. Anyway, I need an unique id for each job. Please, tell me if you know an other way

Reply:
why don't we use `lib/containers/urls.pm`?

Reply:
Sorry, I don't understand what you mean. The problem I would like to avoid is to have two jobs uploading the container image with the same name, because one of these jobs would fail. And would have false positives

Reply:
1) I would pass the name as parameter to the function (with proper default) and let the name by the caller
2) If we use the same image in multiple jobs, it would be nice to reuse it. I think it is time consuming to upload these images. We do the same with vm images in CSP. And then we have a cleanup mechanize in PCW.

can you make an example of the currently used name? Maybe I'm wrong with my assumptions.

Reply:
Now I see, you only extract the job-id. If this image needs to be unique, then you should be unique by openqa instance.
Maybe you could use:
```perl
join('-', get_var('PUBLIC_CLOUD_RESOURCE_NAME'), mmapi::get_current_job_id());
```
But still, I would do it in `tests/containers/upload_container_images_to_ecr.pm`.

Reply:
Understood, good idea

Reply:
done

Reply:
Wait, can you @ilausuch give me an example of the content of `NAME` variable?

Reply:
```
NAME=00000200-sle-15-SP3-Container-Image-Updates-x86_64-Build1-sle-15-SP3_image_on_ECR@64bit
```

Reply:
OK, Good! Thank you :)